tests/erlang_tests/test_utf8_atoms.erl

```diff
@@ -0,0 +1,232 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2024 Davide Bettio <davide@uninstall.it>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(test_utf8_atoms).
+-export([start/0, conv/3, get_atom/1, get_list/1, get_binary/1, conv2/3]).
+
+start() ->
+    32767 - test_from_atom() +
```

Comment:
I find that test arithmetic is not very readable, especially now that we have stack traces and that we run tests with OTP.

Reply:
It's not the best, but the result gives me a bitmask with failed tests, regardless of OTP version. I don't get useful stacktraces with line numbers with all OTP versions.
Anyway if you are strongly opinioned here I can change this with some pattern matching against expected values.